daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/ledgerinteraction/IdeLedgerClient.scala

```diff
@@ -58,12 +61,45 @@ class IdeLedgerClient(
 
   def currentSubmission: Option[ScenarioRunner.CurrentSubmission] = _currentSubmission
 
-  private[this] val preprocessor =
+  private[this] var compiledPackages = originalCompiledPackages
+
+  private[this] var preprocessor = makePreprocessor
+
+  private[this] var disabledPackages: Set[PackageId] = Set.empty
+
+  private[this] def makePreprocessor =
     new preprocessing.CommandPreprocessor(
       compiledPackages.pkgInterface,
       requireV1ContractIdSuffix = false,
     )
 
+  private[this] def partialFunctionFilterNot[A](f: A => Boolean): PartialFunction[A, A] = {
+    case x if !f(x) => x
+  }
+
+  // Given a set of disabled packages, filter out all definitions from those packages from the original compiled packages
+  private[this] def updateCompiledPackages() = {
+    compiledPackages =
+      if (!disabledPackages.isEmpty)(
+        originalCompiledPackages.copy(
+          // Remove packages from the set
+          packageIds = originalCompiledPackages.packageIds -- disabledPackages,
+          // Filter out pkgId "key" to the partial function
+          pkgInterface = new PackageInterface(
+            partialFunctionFilterNot(
+              disabledPackages
+            ) andThen originalCompiledPackages.pkgInterface.signatures
+          ),
+          // Filter out any defs in a disabled package
+          defns = originalCompiledPackages.defns.filter { case (sDefRef, _) =>
+            !disabledPackages(sDefRef.packageId)
+          },
+        ),
+      )
+      else originalCompiledPackages
+    preprocessor = makePreprocessor
+  }
```

Comment:
Is there a nicer way to "remove" a package from CompiledPackages without recompiling fully? I'd assume not given package interdependencies.
Should consider full recompilation, however this is costly and doesn't mesh well with the scenario service (which precompiles itself)